pkg/backend/impl/redis/redis.go

```diff
@@ -290,6 +290,15 @@ func (s *redis) Probe(ctx context.Context) error {
 		return nil
 	}
 
+	info := s.client.Ping(ctx)
+	result, errResult := info.Result()
+	s.logger.Debugw("Probing redis with PING command", zap.String("info", result))
+	if errResult == nil {
+		return nil
+	} else {
+		err = errResult
+	}
+
 	// Add some context since Redis client sometimes is not clear about what failed.
 	return fmt.Errorf("failed probing Redis, retrieving client ID: %w", err)
```

Comment:
sweet, but to be more clear to users, we can log that the first attempt with Client ID failed, then try the ping and log if it fails or not.

```suggestion
        s.logger.Info("Probing redis with CLIENT ID command failed, trying PING command", zap.Error(err))
        
	info := s.client.Ping(ctx)
	result, err := info.Result()
	s.logger.Debugw("Probing redis with PING command", zap.String("info", result))
	if err == nil {
		return nil
	}

	// Add some context since Redis client sometimes is not clear about what failed.
	return fmt.Errorf("failed probing Redis, using PING: %w", err)
```

I also noticed that my previous code was not great, I was always logging this
https://github.com/triggermesh/brokers/blob/5bc3b3202a9d9799729ca91f3c21df3cfaa37059/pkg/backend/impl/redis/redis.go#L287

but I should only when the CLIENT ID command does not return an error. Would you mind modifying that also?

Reply:
Sure no problem, messaging looks clearer now. Thanks!